mlir/include/mlir/IR/OperationSupport.h

```diff
@@ -1271,7 +1271,9 @@ struct OperationEquivalence {
   isEquivalentTo(Operation *lhs, Operation *rhs,
                  function_ref<LogicalResult(Value, Value)> checkEquivalent,
                  function_ref<void(Value, Value)> markEquivalent = nullptr,
-                 Flags flags = Flags::None);
+                 Flags flags = Flags::None,
+                 function_ref<LogicalResult(ValueRange, ValueRange)>
+                     checkCommutativeEquivalent = nullptr);
```

Comment:
This should be documented.

I don't quite get why we need injection here by the way?

Reply:
As this is a static function and we need to be able to inject querying the equivalence classes. Else we have to do N^2 as Matthias suggested.

Reply:
I'm not sure that answers my question, let me rephrase: why isn't this just a Flag?

Reply:
The equivalence classes are not exposed, so only accessible via the functors passed in. If a flag, then one could only access them via checkEquivalent still. Now, an option is to just not care about equivalence for commutative case except for pairwise part.

Reply:
OK, LG, but you're still missing the doc :)